crates/hir-ty/src/diagnostics/match_check/pat_analysis.rs

```diff
@@ -36,6 +36,32 @@ pub(crate) type WitnessPat<'p> = rustc_pattern_analysis::pat::WitnessPat<MatchCh
 #[derive(Copy, Clone, Debug, PartialEq, Eq)]
 pub(crate) enum Void {}
 
+/// An index type for enum variants. This ranges from 0 to `variants.len()`, whereas `EnumVariantId`
+/// can take arbitrary large values (and hence mustn't be used with `IndexVec`/`BitSet`).
+#[derive(Copy, Clone, Debug, PartialEq, Eq, Hash)]
+pub(crate) struct EnumVariantContiguousIndex(usize);
+
+impl EnumVariantContiguousIndex {
+    fn from_enum_variant_id(
+        db: &dyn HirDatabase,
+        eid: EnumId,
+        target_evid: EnumVariantId,
+    ) -> Option<Self> {
+        // Find the index of this variant in the list of variants.
+        db.enum_data(eid)
+            .variants
+            .iter()
+            .map(|(evid, _name)| *evid)
+            .enumerate()
+            .find(|(_, evid)| *evid == target_evid)
+            .map(|(i, _)| EnumVariantContiguousIndex(i))
```

Comment:
This can be simplified to
```suggestion
target_evid.lookup(self.db.upcast()).index as usize
```

(we store the index in the item tree to prevent a linear lookup here)

Reply:
Oh nicee